common/daemonsets_utils.go

```diff
@@ -18,3 +18,32 @@ func NodeSelectorMatchesNodeLabels(node *corev1.Node, nodeSelector *corev1.NodeS
 	matches, err := v1helper.MatchNodeSelectorTerms(node, nodeSelector)
 	return matches, err
 }
+
+func KubeProxySideCar() corev1.Container {
+	return corev1.Container{
+		Name:  "kube-rbac-proxy",
+		Image: GetKubeRBACProxyImage(),
+		Ports: []corev1.ContainerPort{
+			{
+				ContainerPort: int32(9393),
+				Name:          "metrics",
+				Protocol:      corev1.ProtocolTCP,
+			},
+		},
+		Args: []string{
+			"--logtostderr=true",
+			"--secure-listen-address=0.0.0.0:9393",
+			"--upstream=http://127.0.0.1:8383/",
+			"--tls-cert-file=/etc/tls/private/tls.crt",
+			"--tls-private-key-file=/etc/tls/private/tls.key",
+		},
+
+		VolumeMounts: []corev1.VolumeMount{
+			{
+				Name:      metricsServingCert,
+				MountPath: "/etc/tls/private",
+			},
+		},
+	}
+
```

Comment:
nit: can this be moved to daemonset YAML?

Reply:
I thought about that.  But I believe we  might not need this for non openshift environments, right?  So I think we keep it separate from the deployment yaml and add it only if openshift is used. What do you say?

(PS:  we still haven't added check to disabled this for non openshift environments)

Reply:
I am leaning towards, we are kinda breaking plain k8s deployment anyways, so we may as well use this opportunity to keep our code clean. ;-)

Reply:
Hey, sorry for the late comments. How are we breaking k8s deployment?

Reply:
If I'm not wrong, then its the use of `openshift-user-critical` priorityClass (https://bugzilla.redhat.com/show_bug.cgi?id=1967444) that breaks plan k8s. This PR uses `service.beta.openshift.io/serving-cert-secret-name` to create secret for the certs which breaks it further.